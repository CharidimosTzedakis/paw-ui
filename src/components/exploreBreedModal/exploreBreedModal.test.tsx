import { describe, it, Mock, vi, expect } from "vitest";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import theCatAPI from "@api/catApiClient.ts";
import { useLocation } from "wouter";
import ExploreBreedModal from "./exploreBreedModal";
import { catImagesBirmanBreed } from "@test/fixtures";
import type { AvailableBreedsEnumType } from "@api/types";
import { availableBreeds } from "@api/constants/availableBreeds";

vi.mock("@api/catApiClient", () => ({
  default: {
    images: {
      searchImages: vi.fn(),
    },
  },
}));

vi.mock("wouter", async () => {
  const actual = await vi.importActual("wouter");

  return {
    ...actual,
    useLocation: vi.fn(() => ["/current-path", vi.fn()]),
  };
});

describe("<ExploreBreedModal/>", () => {
  it("should render correctly in loading state", () => {
    (theCatAPI.images.searchImages as Mock).mockImplementation(
      () => new Promise(() => {}),
    );

    render(
      <ExploreBreedModal
        breedId={availableBreeds.BIRMAN as AvailableBreedsEnumType}
      />,
    );

    expect(document.body).toMatchSnapshot();
  });

  it("should render correctly a list of images for a specific breed", async () => {
    (theCatAPI.images.searchImages as Mock).mockResolvedValueOnce(
      catImagesBirmanBreed,
    );

    render(
      <ExploreBreedModal
        breedId={availableBreeds.BIRMAN as AvailableBreedsEnumType}
      />,
    );

    await waitFor(() => {
      expect(screen.queryByText("Image placeholder")).not.toBeInTheDocument();
    });

    expect(document.body).toMatchSnapshot();
  });

  it("should navigate to  parent route when the 'close' button is clicked", async () => {
    const setLocation = vi.fn();
    (useLocation as Mock).mockReturnValue(["/current-path", setLocation]);
    (theCatAPI.images.searchImages as Mock).mockResolvedValueOnce(
      catImagesBirmanBreed,
    );

    render(
      <ExploreBreedModal
        breedId={availableBreeds.BIRMAN as AvailableBreedsEnumType}
      />,
    );

    await waitFor(() => {
      expect(screen.queryByText("Image placeholder")).not.toBeInTheDocument();
    });

    const closeButton = screen.getByLabelText("Close");
    await userEvent.click(closeButton);

    expect(setLocation).toHaveBeenCalledTimes(1);
    expect(setLocation).toHaveBeenCalledWith("/");
  });
});
