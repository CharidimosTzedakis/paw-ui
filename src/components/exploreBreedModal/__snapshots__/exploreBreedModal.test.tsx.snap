// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ExploreBreedModal/> > should render correctly a list of images for a specific breed 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-1kf000u"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-1kf000u ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="top: 20px; width: 800px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-header"
              >
                <div
                  class="ant-modal-title"
                  id="test-id"
                >
                  <h2
                    class="ant-typography _exploreBreedModalTitle_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    Explore BreedÂ 
                    Birman
                  </h2>
                  <span
                    class="ant-typography css-dev-only-do-not-override-1kf000u"
                  >
                    The following cats are from the breed. Click on them for more info!
                  </span>
                </div>
              </div>
              <div
                class="ant-modal-body"
              >
                <div
                  class="_exploreBreedModalContent_91e42f"
                >
                  <a
                    href="../../cats/OhTkBTPnD"
                  >
                    <div
                      class="ant-image css-dev-only-do-not-override-1kf000u"
                    >
                      <img
                        class="ant-image-img css-dev-only-do-not-override-1kf000u"
                        src="https://cdn2.thecatapi.com/images/OhTkBTPnD.jpg"
                      />
                    </div>
                  </a>
                  <a
                    href="../../cats/Dm0H1zSK1"
                  >
                    <div
                      class="ant-image css-dev-only-do-not-override-1kf000u"
                    >
                      <img
                        class="ant-image-img css-dev-only-do-not-override-1kf000u"
                        src="https://cdn2.thecatapi.com/images/Dm0H1zSK1.jpg"
                      />
                    </div>
                  </a>
                  <a
                    href="../../cats/xRMeDCybn"
                  >
                    <div
                      class="ant-image css-dev-only-do-not-override-1kf000u"
                    >
                      <img
                        class="ant-image-img css-dev-only-do-not-override-1kf000u"
                        src="https://cdn2.thecatapi.com/images/xRMeDCybn.jpg"
                      />
                    </div>
                  </a>
                  <a
                    href="../../cats/3rK7HGo7r"
                  >
                    <div
                      class="ant-image css-dev-only-do-not-override-1kf000u"
                    >
                      <img
                        class="ant-image-img css-dev-only-do-not-override-1kf000u"
                        src="https://cdn2.thecatapi.com/images/3rK7HGo7r.jpg"
                      />
                    </div>
                  </a>
                  <a
                    href="../../cats/2LEN_GHmx"
                  >
                    <div
                      class="ant-image css-dev-only-do-not-override-1kf000u"
                    >
                      <img
                        class="ant-image-img css-dev-only-do-not-override-1kf000u"
                        src="https://cdn2.thecatapi.com/images/2LEN_GHmx.jpg"
                      />
                    </div>
                  </a>
                  <a
                    href="../../cats/C0BNiXO0T"
                  >
                    <div
                      class="ant-image css-dev-only-do-not-override-1kf000u"
                    >
                      <img
                        class="ant-image-img css-dev-only-do-not-override-1kf000u"
                        src="https://cdn2.thecatapi.com/images/C0BNiXO0T.jpg"
                      />
                    </div>
                  </a>
                  <a
                    href="../../cats/znCxQCEp4"
                  >
                    <div
                      class="ant-image css-dev-only-do-not-override-1kf000u"
                    >
                      <img
                        class="ant-image-img css-dev-only-do-not-override-1kf000u"
                        src="https://cdn2.thecatapi.com/images/znCxQCEp4.jpg"
                      />
                    </div>
                  </a>
                  <a
                    href="../../cats/9tZCXEYzy"
                  >
                    <div
                      class="ant-image css-dev-only-do-not-override-1kf000u"
                    >
                      <img
                        class="ant-image-img css-dev-only-do-not-override-1kf000u"
                        src="https://cdn2.thecatapi.com/images/9tZCXEYzy.jpg"
                      />
                    </div>
                  </a>
                  <a
                    href="../../cats/TKKtEJZVl"
                  >
                    <div
                      class="ant-image css-dev-only-do-not-override-1kf000u"
                    >
                      <img
                        class="ant-image-img css-dev-only-do-not-override-1kf000u"
                        src="https://cdn2.thecatapi.com/images/TKKtEJZVl.jpg"
                      />
                    </div>
                  </a>
                  <a
                    href="../../cats/yZE2JpeXz"
                  >
                    <div
                      class="ant-image css-dev-only-do-not-override-1kf000u"
                    >
                      <img
                        class="ant-image-img css-dev-only-do-not-override-1kf000u"
                        src="https://cdn2.thecatapi.com/images/yZE2JpeXz.jpg"
                      />
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`<ExploreBreedModal/> > should render correctly in loading state 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-1kf000u"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-1kf000u ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="top: 20px; width: 800px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-header"
              >
                <div
                  class="ant-modal-title"
                  id="test-id"
                >
                  <h2
                    class="ant-typography _exploreBreedModalTitle_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    Explore BreedÂ 
                    Birman
                  </h2>
                  <span
                    class="ant-typography css-dev-only-do-not-override-1kf000u"
                  >
                    The following cats are from the breed. Click on them for more info!
                  </span>
                </div>
              </div>
              <div
                class="ant-modal-body"
              >
                <div
                  class="_exploreBreedModalContent_91e42f"
                >
                  <div
                    class="ant-skeleton ant-skeleton-element ant-skeleton-active _exploreBreedModalSkeleton_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    <div
                      class="ant-skeleton-image _exploreBreedModalSkeleton_91e42f"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Image placeholder
                        </title>
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="ant-skeleton ant-skeleton-element ant-skeleton-active _exploreBreedModalSkeleton_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    <div
                      class="ant-skeleton-image _exploreBreedModalSkeleton_91e42f"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Image placeholder
                        </title>
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="ant-skeleton ant-skeleton-element ant-skeleton-active _exploreBreedModalSkeleton_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    <div
                      class="ant-skeleton-image _exploreBreedModalSkeleton_91e42f"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Image placeholder
                        </title>
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="ant-skeleton ant-skeleton-element ant-skeleton-active _exploreBreedModalSkeleton_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    <div
                      class="ant-skeleton-image _exploreBreedModalSkeleton_91e42f"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Image placeholder
                        </title>
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="ant-skeleton ant-skeleton-element ant-skeleton-active _exploreBreedModalSkeleton_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    <div
                      class="ant-skeleton-image _exploreBreedModalSkeleton_91e42f"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Image placeholder
                        </title>
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="ant-skeleton ant-skeleton-element ant-skeleton-active _exploreBreedModalSkeleton_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    <div
                      class="ant-skeleton-image _exploreBreedModalSkeleton_91e42f"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Image placeholder
                        </title>
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="ant-skeleton ant-skeleton-element ant-skeleton-active _exploreBreedModalSkeleton_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    <div
                      class="ant-skeleton-image _exploreBreedModalSkeleton_91e42f"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Image placeholder
                        </title>
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="ant-skeleton ant-skeleton-element ant-skeleton-active _exploreBreedModalSkeleton_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    <div
                      class="ant-skeleton-image _exploreBreedModalSkeleton_91e42f"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Image placeholder
                        </title>
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="ant-skeleton ant-skeleton-element ant-skeleton-active _exploreBreedModalSkeleton_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    <div
                      class="ant-skeleton-image _exploreBreedModalSkeleton_91e42f"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Image placeholder
                        </title>
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="ant-skeleton ant-skeleton-element ant-skeleton-active _exploreBreedModalSkeleton_91e42f css-dev-only-do-not-override-1kf000u"
                  >
                    <div
                      class="ant-skeleton-image _exploreBreedModalSkeleton_91e42f"
                    >
                      <svg
                        class="ant-skeleton-image-svg"
                        viewBox="0 0 1098 1024"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          Image placeholder
                        </title>
                        <path
                          class="ant-skeleton-image-path"
                          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
